protocol/network/3-requirements.md

```diff
@@ -0,0 +1,104 @@
+3 Requirements and considerations
+=================================
+
+## 3.1 NAT traversal
+
+Network Address Translation is ubiquitous in the internet. Not only are most consumer devices behind many layers of NATs, but most datacenter nodes are often behind NAT for security or virtualization reasons. As we move into containerized deployments, this is getting worse. IPFS implementations SHOULD provide a way to traverse NATs, otherwise it is likely that operation will be affected. Even nodes meant to run with real IP addresses must implement NAT traversal techniques, as they may need to establish connections to peers behind NAT.
+
+libp2p accomplishes full NAT traversal using an ICE-like protocol. It is not exactly ICE, as ipfs networks provide the possibility of relaying communications over the IPFS protocol itself, for coordinating hole-punching or even relaying communication.
+
+It is recommended that implementations use one of the many NAT traversal libraries available, such as `libnice`, `libwebrtc`, or `natty`. However, NAT traversal must be interoperable.
+
+## 3.2 Relay
+
+Unfortunately, due to symmetric NATs, container and VM NATs, and other impossible-to-bypass NATs, libp2p MUST fallback to relaying communication to establish a full connectivity graph. To be complete, implementations MUST support relay, though it SHOULD be optional and able to be turned off by end users.
+
+## 3.3 Encryption
+
+Communications on libp2p may be:
+
+- **encrypted**
+- **signed** (not encrypted)
+- **clear** (not encrypted, not signed)
+
+We take both security and performance seriously. We recognize that encryption is not viable for some in-datacenter high performance use cases.
+
+We recommend that:
+- implementations encrypt all communications by default
+- implementations are audited
+- unless absolutely necessary, users normally operate with encrypted communications only.
+
+libp2p uses cyphersuites like TLS.
+
+**NOTE:** we do not use lib2p directly, because we do not want the CA system baggage. Most libp2p implementations are very big. Since the lib2p model begins with keys, libp2p only needs to apply ciphers. This is a minimal portion of the whole TLS standard.
```

Comment:
This doesn't hold true today and must be updated. Discussion - https://github.com/ipfs/specs/issues/29